display/more.rs

```diff
@@ -0,0 +1,2952 @@
+//
+// Copyright (c) 2024 Jeff Garzik
+//
+// This file is part of the posixutils-rs project covered under
+// the MIT License.  For the full license text, please see the LICENSE
+// file in the root directory of this project.
+// SPDX-License-Identifier: MIT
+//
+
+use clap::Parser;
+use gettextrs::{bind_textdomain_codeset, setlocale, textdomain, LocaleCategory};
+use libc::{
+    getegid, getgid, getuid, regcomp, regex_t, regexec, setgid, setuid, REG_ICASE, REG_NOMATCH,
+};
+use std::collections::HashMap;
+use std::ffi::CString;
+use std::fs::File;
+use std::io::{stdout, BufRead, BufReader, Cursor, Read, Seek, SeekFrom, Write};
+use std::ops::{Not, Range};
+use std::os::fd::AsRawFd;
+use std::path::PathBuf;
+use std::process::{exit, ExitStatus};
+use std::ptr;
+use std::str::FromStr;
+use std::sync::mpsc::{channel, Receiver, TryRecvError};
+use std::sync::Mutex;
+use std::time::Duration;
+use termion::{clear::*, cursor::*, event::*, input::*, raw::*, screen::*, style::*, *};
+
+const LINES_PER_PAGE: u16 = 24;
+const NUM_COLUMNS: u16 = 80;
+const DEFAULT_EDITOR: &str = "vi";
+const CONVERT_STRING_BUF_SIZE: usize = 64;
+const PROJECT_NAME: &str = "posixutils-rs";
+
+/// Last acceptable pressed mouse button
+static LAST_MOUSE_BUTTON: Mutex<Option<MouseButton>> = Mutex::new(None);
+/// Inform terminal input handler thread that program is closing
+static NEED_QUIT: Mutex<bool> = Mutex::new(false);
+
+/// more - display files on a page-by-page basis.
+#[derive(Parser)]
+#[command(version, about = "more - display files on a page-by-page basis")]
+struct Args {
+    // Enable interactive session test
+    #[arg(long = "test", help = "Enable interactive session test.")]
+    test: bool,
+
+    /// Do not scroll, display text and clean line ends
+    #[arg(short = 'c', help = "Do not scroll, display text and clean line ends.")]
+    print_over: bool,
+
+    /// Exit on end-of-file
+    #[arg(short = 'e', help = "Exit on end-of-file.")]
+    exit_on_eof: bool,
+
+    /// Perform pattern matching in searches without regard to case
+    #[arg(
+        short = 'i',
+        help = "Perform pattern matching in searches without regard to case."
+    )]
+    case_insensitive: bool,
```

Comment:
Align long args with known utils such as GNU linux:
```
Options:
 -d, --silent          display help instead of ringing bell
 -f, --logical         count logical rather than screen lines
 -l, --no-pause        suppress pause after form feed
 -c, --print-over      do not scroll, display text and clean line ends
 -p, --clean-print     do not scroll, clean screen and display text
 -s, --squeeze         squeeze multiple blank lines into one
 -u, --plain           suppress underlining and bold
 -n, --lines <number>  the number of lines per screenful
 -<number>             same as --lines
 +<number>             display file beginning from line number
 +/<pattern>           display file beginning from pattern match

```